### DISCUSSION TITLE
Size of uart `rx_buffer_size`

### DISCUSSION THREAD

**Discussion by Sleeper85** [syssi/esphome-jk-bms#605] [General]
Hi @syssi 

For my YamBMS project where UART ports are preconfigured, I would like to increase the `rx_buffer_size` value to `1024`.

What would be the impact of doing this with your `jk_bms` component over UART ?

The goal would be to have a buffer size that can work with both JK-B and JK-PB (RS485).

Thanks for your help

**Comment by syssi** [comment]
IMO there is no impact. The cell info frame must fit into the buffer (once!). If you increase the buffer the memory will be reserved but never used.

**Reply by Sleeper85** [reply]
And if the buffer is too small, what happens ?

@txubelaxu RS485 component uses a `rx_buffer` of `1000` by default.
Unfortunately, I don't have a JK-PB to do the tests. I wonder if his component could work with a `rx_buffer` of `512`?

I have someone who uses my project with 7x JK-PB, I can ask him to do some tests.
I also wonder if having 7x JK-PB requires a bigger `rx_buffer` than with 1x JK-PB ?

I would like to find a value that can fit both components without choosing `1024` if it is not necessary.

**Reply by syssi** [reply]
If the RX buffer is too small the component won't be able to decode incoming frames (which exceeds the size of the buffer).

**Reply by Sleeper85** [reply]
Ok so the problem will be visible directly, it remains to be seen whether the size of this frame increases with the number of BMS used (a single UART connection with RS485).

**Reply by syssi** [reply]
I don't understand your last statement. Why do you expect the frame length increases with the number of devices?

**Reply by Sleeper85** [reply]
This is a question I ask myself.

Does the size of this frame increase with the number of BMS to supervise? Retrieving information from 7x JK-PB (on the same RS485 links) is 7x more information. But maybe @txubelaxu code makes a request for each BMS and the frame keeps the same size...

I don't have a JK-PB to check this, the easiest thing is to ask someone to test it for me.
